include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -27,6 +27,27 @@ namespace oneapi
 {
 namespace dpl
 {
+
+namespace __internal
+{
+
+template <typename _R>
```

Comment:
Since `_R` is a public API do we want to add `static_assert` here for meaningful error if both overloads are ill-formed.

Reply:
probably, it makes sense

Reply:
But., i doesn't know how to write the third case (for other types), because here (and in other same places)  we have just two "overloads" `get_value_type(int)` and `get_value_type(...)`... or we have to re-write "SFINAE/overloads" approach....